core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -1374,35 +1374,44 @@ class KafkaApis(val requestChannel: RequestChannel,
     val initPidRequest = request.body[InitPidRequest]
     val transactionalId = initPidRequest.transactionalId
 
-    // Send response callback
-    def sendResponseCallback(result: InitPidResult): Unit = {
-      def createResponse(throttleTimeMs: Int): AbstractResponse = {
-        val responseBody: InitPidResponse = new InitPidResponse(throttleTimeMs, result.error, result.pid, result.epoch)
-        trace(s"InitPidRequest: Completed $transactionalId's InitPidRequest with result $result from client ${request.header.clientId}.")
-        responseBody
+    if(authorize(request.session, Write, new Resource(ProducerTransactionalId, transactionalId))) {
```

Comment:
@hachikuji there is no PID within the produce request. Do we need to add one? I.e, if the producer isn't idempotent there won't ever be a PID - right?